install/kubernetes/cilium/templates/cilium-agent/daemonset.yaml

```diff
@@ -261,13 +261,13 @@ spec:
           containerPort: {{ .Values.prometheus.port }}
           hostPort: {{ .Values.prometheus.port }}
           protocol: TCP
-        {{- if and .Values.envoy.prometheus.enabled (not $envoyDS) }}
+        {{- if and .Values.envoy.enabled .Values.envoy.prometheus.enabled (not $envoyDS) }}
```

Comment:
The Helm flag `envoy.enabled` controls whether Envoy should be deployed as DaemonSet - and NOT whether Envoy should be enabled in general. Unfortunately, the naming of that flag isn't ideal. 

IMO this change doesn't make sense - or I don't get the actual idea of it. 

https://github.com/cilium/cilium/blob/main/install/kubernetes/cilium/values.yaml.tmpl#L2386-L2389

Reply:
Makes sense - thanks!